### PR TITLE
app/eth2wrap: sign domain for `VOLUNTARY_EXIT`

### LINKED ISSUE #2981: Ensure Exit Message are compliant with EIP-7044
## 🎯 Problem to be solved
EIP-7044 mandates exit messages to be signed on Capella's domain. In Charon, exit messages aren't signed, but their validity is verified, typically using the last available domain. It's crucial to ensure the domain from Capella is used for this purpose.

## 🛠️ Proposed solution
- [ ] Implement a mechanism to identify and use Capella's domain for signing exit messages in compliance with EIP-7044.
- [ ] Update the validity verification process in Charon to prioritize Capella's domain when checking exit messages.

### PR DESCRIPTION
**Pull request by gsora**
Instead of returning the latest Domain available when the DomainName requested is VOLUNTARY_EXIT, returns the Capella one as detailed in EIP-7044.

Doing so fixes both `exit` commands generating wrong signatures and voluntary exits sent over the VC.

category: bug
ticket: #2981

Closes #2981.

### CODE REVIEW

**Review by pinebit — APPROVED** [review]
It may be hard to merge this with the v3 endpoint branch...

**Comment by gsora** [comment]
@pinebit why do you think that? Do you believe we should hold off this PR until V3 is merged?

**Comment by pinebit** [comment]
> @pinebit why do you think that? Do you believe we should hold off this PR until V3 is merged?

@gsora no worries, I will handle the merge.

**Pull request merged**
